src/OpenSage.Game/Logic/Object/Draw/W3dModelDraw.cs

```diff
@@ -172,10 +172,15 @@ private W3dModelDrawConditionState CreateModelDrawConditionStateInstance(ModelCo
             {
                 foreach (var particleSysBone in conditionState.ParticleSysBones)
                 {
-                    var particleSystemTemplate = _contentManager.IniDataContext.FXParticleSystems.First(x => x.Name == particleSysBone.ParticleSystem);
+                    var particleSystemTemplate = _contentManager.IniDataContext.FXParticleSystems.FirstOrDefault(x => x.Name == particleSysBone.ParticleSystem);
                     if (particleSystemTemplate == null)
                     {
-                        particleSystemTemplate = _contentManager.IniDataContext.ParticleSystems.First(x => x.Name == particleSysBone.ParticleSystem).ToFXParticleSystemTemplate();
+                        particleSystemTemplate = _contentManager.IniDataContext.ParticleSystems.FirstOrDefault(x => x.Name == particleSysBone.ParticleSystem).ToFXParticleSystemTemplate();
+
+                        if (particleSystemTemplate == null)
+                        {
+                            throw new InvalidOperationException("Missing referenced particle system: " + particleSysBone.ParticleSystem);
```

Comment:
This is fine for now, although we might want to convert this into a warning when we get around implementing a logging system.